Essentials/src/main/java/com/earth2me/essentials/Mail.java

```diff
@@ -0,0 +1,112 @@
+package com.earth2me.essentials;
+
+import com.earth2me.essentials.messaging.IMessageRecipient;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import static com.earth2me.essentials.I18n.tl;
+
+public class Mail {
+    private final transient IEssentials ess;
+    private final transient JsonParser jsonParser = new JsonParser();
+    private final transient ThreadLocal<SimpleDateFormat> df = ThreadLocal.withInitial(() -> new SimpleDateFormat("yyyy/MM/dd HH:mm"));
+
+    public Mail(IEssentials ess) {
+        this.ess = ess;
+    }
+
+    public void sendMail(IUser user, IMessageRecipient sender, String message) {
+        sendMail(user, sender, message, 0);
+    }
+
+    public void sendMail(IUser user, IMessageRecipient sender, String message, long expireAt) {
+        final JsonArray jsonArray = new JsonArray();
+        final JsonObject mail = new JsonObject();
+        mail.addProperty("read", false);
+        mail.addProperty("sender", sender.getName());
+        mail.addProperty("uuid", sender.getUuid() == null ? "null" : sender.getUuid().toString());
+        mail.addProperty("timestamp", System.currentTimeMillis());
+        mail.addProperty("expire", expireAt);
+        mail.addProperty("message", message);
+        jsonArray.add(mail);
```

Comment:
Would much prefer we used a plain/immutable `MailMessage`/`Mail.Message` class for individual mails' data. This would let us object map + expose a much nicer mails API. Also, I'm not a huge fan of shoving JSON inside a string in another YAML file; may want to consider storing mails in a `mails.json` or something similar instead.